mkosi/distributions/opensuse.py

```diff
@@ -21,47 +23,128 @@ def filesystem(cls) -> str:
 
     @classmethod
     def install(cls, state: "MkosiState") -> None:
+        if state.config.base_image:
+            # We assume that the base image has been properly initialized and it
+            # contains all the metadata we need to install the additional
+            # packages.
+            return zypper_install(state, {*state.config.packages})
+
         return install_opensuse(state)
 
+    @classmethod
+    def remove_packages(cls, state: "MkosiState", packages: list[str]) -> None:
+        zypper_remove(state, packages)
+
     @staticmethod
     def initrd_path(kver: str) -> Path:
         return Path("boot") / f"initrd-{kver}"
 
 
+def invoke_zypper(state: MkosiState,
+                  global_opts: list[str],
+                  verb: str,
+                  verb_opts: list[str],
+                  *args: Any,
+                  with_apivfs: bool = False) -> None:
+
+    cmdline = ["zypper", "--root", state.root, *global_opts, verb, *verb_opts, *args]
+    env={"ZYPP_CONF": state.root.joinpath("etc/zypp/zypp.conf")}
+
+    if with_apivfs:
+        run_with_apivfs(state, cmdline, env=env)
+    else:
+        run(cmdline, env=env)
+
+
+def zypper_init(state: MkosiState) -> None:
+    state.root.joinpath("etc/zypp").mkdir(mode=0o755, parents=True, exist_ok=True)
+
+    # No matter if --root is used or not, zypper always considers its config
+    # files from the host environment. If we want to use our custom versions for
+    # the rootfs, we're left with two ways to specify them, depending on whether
+    # we need to customize a setting defined in zypp.conf or in zypper.conf. If
+    # it's in zypp.conf then the environment variable 'ZYPP_CONF' must be used
+    # (!) otherwise a custom zypper.conf can be specified with the global option
+    # '--config'.
+
+    zypp_conf = state.root.joinpath("etc/zypp/zypp.conf")
+
+    # For some reason zypper has no command line option to exclude the docs,
+    # this can only be configured via zypp.conf.
+    def yes_no(b: bool) -> str:
+        return "yes" if b else "no"
+
+    zypp_conf.write_text(
+        dedent(
+             f"""\
+             [main]
+             solver.onlyRequires = yes
+             rpm.install.excludedocs = {yes_no(not state.config.with_docs)}
+             """
```

Comment:
```suggestion
    zypp_conf.write_text(
        dedent(
             f"""\
             [main]
             solver.onlyRequires = yes
             rpm.install.excludedocs = {"no" if state.config.with_docs else "yes"}
```
can we inline this to avoid the double negative?